trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/map-to-selected-payment-details-filters-view-model.test.ts

```diff
@@ -0,0 +1,142 @@
+
+      // Act
+      const result = mapToSelectedPaymentDetailsFiltersViewModel(filters, reportId);
+
+      // Assert
+      expect(result!.facilityId!.value).toEqual('12345');
+    });
+
+    it('should set paymentCurrency selected filter value to the payment currency when payment currency provided', () => {
+      // Arrange
+      const filters = { paymentCurrency: CURRENCY.JPY };
+
+      // Act
+      const result = mapToSelectedPaymentDetailsFiltersViewModel(filters, reportId);
+
+      // Assert
+      expect(result!.paymentCurrency!.value).toEqual(CURRENCY.JPY);
+    });
+
+    it('should set paymentReference selected filter value to the payment reference when payment reference provided', () => {
+      // Arrange
+      const filters = { paymentReference: 'REF123' };
+
+      // Act
+      const result = mapToSelectedPaymentDetailsFiltersViewModel(filters, reportId);
+
+      // Assert
+      expect(result!.paymentReference!.value).toEqual('REF123');
+    });
+
+    it('should set facilityId remove href to link to report with other active filters still applied', () => {
+      // Arrange
+      const filters = { facilityId: '12345', paymentReference: 'REF123', paymentCurrency: CURRENCY.USD };
+
+      when(getPaymentDetailsTabHref)
+        .calledWith({ paymentReference: 'REF123', paymentCurrency: CURRENCY.USD }, reportId)
+        .mockReturnValue('/remove-facility-id-filter');
+
+      // Act
+      const result = mapToSelectedPaymentDetailsFiltersViewModel(filters, reportId);
+
+      // Assert
+      expect(result!.facilityId!.removeHref).toEqual('/remove-facility-id-filter');
+    });
+
+    it('should set paymentCurrency remove href to link to report with other active filters still applied', () => {
+      // Arrange
+      const filters = { facilityId: '12345', paymentReference: 'REF123', paymentCurrency: CURRENCY.USD };
+
+      when(getPaymentDetailsTabHref)
+        .calledWith({ facilityId: '12345', paymentReference: 'REF123' }, reportId)
+        .mockReturnValue('/remove-payment-currency-filter');
+
+      // Act
+      const result = mapToSelectedPaymentDetailsFiltersViewModel(filters, reportId);
+
+      // Assert
+      expect(result!.paymentCurrency!.removeHref).toEqual('/remove-payment-currency-filter');
+    });
+
+    it('should set paymentReference remove href to link to report with other active filters still applied', () => {
+      // Arrange
+      const filters = { facilityId: '12345', paymentReference: 'REF123', paymentCurrency: CURRENCY.USD };
+
+      when(getPaymentDetailsTabHref)
+        .calledWith({ facilityId: '12345', paymentCurrency: CURRENCY.USD }, reportId)
+        .mockReturnValue('/remove-payment-reference-filter');
+
+      // Act
+      const result = mapToSelectedPaymentDetailsFiltersViewModel(filters, reportId);
+
+      // Assert
+      expect(result!.paymentReference!.removeHref).toEqual('/remove-payment-reference-filter');
+    });
+  });
+});
```

Comment:
Can we test the full return pleae?

Reply:
I've split it so that we have separate assertions in separate tests for the different things that can individually go wrong, we lose that specificity if we test the full return in one.

Reply:
Hmm I kind of see your point, we do have a return type on the function however so we would get a linting error if we accidentally threw in extra fields:
<img width="1120" alt="Screenshot 2024-10-23 at 17 35 28" src="https://github.com/user-attachments/assets/4d34a3de-2996-4983-a209-55d716db9136">